src/opm/io/eclipse/EGrid.cpp

```diff
@@ -28,43 +30,188 @@
 #include <string>
 #include <sstream>
 
+#include <math.h>
+
 namespace Opm { namespace EclIO {
 
-EGrid::EGrid(const std::string &filename) : EclFile(filename)
+using NNCentry = std::tuple<int, int, int, int, int,int, float>;
+
+EGrid::EGrid(const std::string &filename, std::string grid_name) :
+    EclFile(filename), inputFileName { filename }, m_grid_name {grid_name}
 {
-   EclFile file(filename);
+    initFileName = inputFileName.parent_path() / inputFileName.stem();
+
+    if (this->formattedInput())
+        initFileName += ".FINIT";
+    else
+        initFileName += ".INIT";
+
+    std::string lgrname = "global";
 
-   auto gridhead = get<int>("GRIDHEAD");
+    m_nncs_loaded = false;
+    actnum_array_index = -1;
+    nnc1_array_index = -1;
+    nnc2_array_index = -1;
 
-   nijk[0] = gridhead[1];
-   nijk[1] = gridhead[2];
-   nijk[2] = gridhead[3];
+    int hostnum_index = -1;
+
+    for (size_t n = 0; n < array_name.size(); n++) {
+
+        if (array_name[n] == "ENDLGR")
+            lgrname = "global";
+
+        if (array_name[n] == "LGR") {
+            auto lgr = this->get<std::string>(n);
+            lgrname = lgr[0];
+            lgr_names.push_back(lgr[0]);
+        }
+
+        if (array_name[n] == "NNCHEAD"){
+            auto nnchead = this->get<int>(n);
+
+            if (nnchead[1] == 0)
+               lgrname = "global";
+            else
+               lgrname = lgr_names[nnchead[1] - 1];
+        }
+
+        if (lgrname == grid_name) {
+            if (array_name[n] == "GRIDHEAD") {
+                auto gridhead = get<int>(n);
+                nijk[0] = gridhead[1];
+                nijk[1] = gridhead[2];
+                nijk[2] = gridhead[3];
+
+                m_radial = gridhead[26] > 0 ? true: false;
+            }
+
+            if (array_name[n] == "COORD")
+                coord_array_index= n;
+            if (array_name[n] == "ZCORN")
```

Comment:
it cannot be all of these, use else to avoid unnecessary string comparisons.

Reply:
you are correct. using else if.